### PR TITLE
[enhance](mtmv)Limit the number of partitions for table creation

### PR DESCRIPTION
**Pull request by zddr**
## Proposed changes

Issue Number: close #xxx

- Creating too many partitions is time-consuming, so limiting the number of partitions
- add more case,such as `mor`,`mow`
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/PartitionDesc.java [hunk 1]
```diff
@@ -159,6 +159,15 @@ public void analyze(List<ColumnDef> columnDefs, Map<String, String> otherPropert
             throw new AnalysisException("No partition columns.");
         }
 
+        int createTablePartitionMaxNum = ConnectContext.get().getSessionVariable().getCreateTablePartitionMaxNum();
+        if (singlePartitionDescs.size() > createTablePartitionMaxNum) {
+            throw new org.apache.doris.nereids.exceptions.AnalysisException(String.format(
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/PartitionDesc.java:164]: should use legacy planner's AnalysisException

**Review by morrySnow — DISMISSED** [review]

**Comment by zddr** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by seawinde — APPROVED** [review]

**Pull request merged**